ODIN_II/SRC/ast_elaborate.cpp

```diff
@@ -2072,6 +2072,9 @@ ast_node_t* reduce_expressions(ast_node_t* node, sc_hierarchy* local_ref, long*
                             oassert(node_is_constant(var_node->children[5]) && node_is_constant(var_node->children[6]));
                         }
 
+                        if (assignment_size > 0) {
+                            var_node->children[1]->types.vnumber = new VNumber(assignment_size - 1);
```

Comment:
what if there is already something there!! you are just overwriting everything here, you cant do that. you need to be specific

Reply:
> what if there is already something there!!

Yes, you are right. It's absolutely not specific!

Reply:
Would you please explain how we can change it specifically? Since even in AST_elaboration we are again coping with recursive function!
Also, is there any reason that the types.hierarchy pointer has not been set in this part?

Reply:
closing this PR. you can open a new one once you have a solution